pkg/services/ngalert/writer/prom.go

```diff
@@ -94,7 +180,55 @@ func (w PrometheusWriter) Write(ctx context.Context, name string, t time.Time, f
 		return err
 	}
 
-	// TODO: placeholder for actual remote write call
-	l.Debug("writing points", "points", points)
+	series := make([]promremote.TimeSeries, 0, len(points))
+	for _, p := range points {
+		series = append(series, promremote.TimeSeries{
+			Labels: promremoteLabelsFromPoint(p),
+			Datapoint: promremote.Datapoint{
+				Timestamp: p.Metric.T,
+				Value:     p.Metric.V,
+			},
+		})
+	}
+
+	l.Debug("Writing metric", "name", name)
+	_, writeErr := w.client.WriteTimeSeries(ctx, series, promremote.WriteOptions{})
```

Comment:
This write has traces and logs, but it could use a metric to alert on write failures. Could we add a histogram measuring the write latency + a counter with a status code label to measure failures? Something like [this](https://github.com/grafana/grafana-enterprise/blob/a64bb0b41a7a1add75f1002e2a20f4af68a1d477/src/pkg/extensions/remoteruler/metrics.go#L34-L46)

Reply:
Could also be a layer up if you'd prefer, but i think there's some merit in having a metric per concrete writer type (or making the writer type a label)

Reply:
I agree, went with writer type as a label 👍